src/Generator.cpp

```diff
@@ -985,6 +990,25 @@ int generate_filter_main_inner(int argc, char **argv, std::ostream &error_output
                                                               json_compiler_logger_factory :
                                                               no_compiler_logger_factory;
 
+    std::atomic<bool> generator_finished = false;
+    int timeout_in_seconds = std::stoi(flags_info["-t"]);
+    // Use ~1 year for "infinite" timeouts.
+    if (timeout_in_seconds <= 0) {
+        timeout_in_seconds = 60 * 24 * 365;
+    }
+    const auto timeout_time = std::chrono::steady_clock::now() + std::chrono::seconds(timeout_in_seconds);
+    std::thread timeout_monitor([timeout_time, timeout_in_seconds, &generator_finished]() {
+        while (!generator_finished) {
```

Comment:
Why not just put `if (timeout_in_seconds <= 0) { return; }` to the lambda?

Reply:
Done